addons/binding/org.openhab.binding.harmonyhub/src/main/java/org/openhab/binding/harmonyhub/handler/HarmonyHubHandler.java

```diff
@@ -388,23 +381,23 @@ public void pressButton(String device, String button) {
         }
     }
 
-    /**
-     * Cache a call to get the harmony config so we don't overwhelm it with requests.
-     *
-     * @return
-     */
-    public synchronized HarmonyConfig getCachedConfig() {
-        if (client == null) {
-            return null;
-        }
+    public CompletableFuture<HarmonyConfig> getConfigFuture() {
+        Supplier<HarmonyConfig> configSupplier = () -> {
+            if (client == null) {
+                throw new IllegalStateException("Client is null");
+            }
+            try {
+                logger.debug("Getting config from client");
+                return client.getConfig();
+            } catch (Exception e) {
```

Comment:
Are you sure this still adds value, you rethrow the exception which you just caught, I would suppose that should only lead to duplicated logs, or is the exception of this supplier silently swallowed somewhere?

Reply:
The current code has been adapted to not swallow the exceptions anywhere. I.e. each call to `getConfigFuture()` is also followed with `.exceptionally(..)`. 

The added value this catch/log/rethrow is:
* It will also log a full stacktrace on debug when an exception occurs. Otherwise the stracktrace  gets lost because it is typically handled by updating the Thing status with just the exception message.
* It is easy to forget adding an `.exceptionally(..)` when new code is created using `getConfigFuture()`. So this would result in a swallowed exception. I hope to prevent countless hours of debugging by always logging the exception here. :slightly_smiling_face: The logged exception should trigger people to also add an `.exceptionally(..)` to properly handle exceptions.

Reply:
I don't no whether I agree but there are enough proper considerations :-)